examples/expressions/13_timeseries_forecasting.py

```diff
@@ -0,0 +1,527 @@
+        lagged_min_24h=pl.col("cnt").shift(1).rolling_min(24),
+        lagged_mean_7d=pl.col("cnt").shift(1).rolling_mean(7 * 24),
+        lagged_max_7d=pl.col("cnt").shift(1).rolling_max(7 * 24),
+        lagged_min_7d=pl.col("cnt").shift(1).rolling_min(7 * 24),
+    )
+    return lagged_df
+
+
+# %%
+# To add lagged features to the data, we go back to the original ``data_var``, and
+# drop the columns that are not relevant. We will have to clean the data again.
+
+
+data_prep = data_var.drop("casual", "instant", "registered").skb.mark_as_X()
+data_prep = data_prep.skb.apply(Cleaner())
+
+
+# %%
+# We can use the |apply_func| method to apply the lagged feature function
+# to the data. The lagged features are added to the dataframe, and we can then
+# concatenate them with the original data. This is done by using the ``.skb.concat``
+# method, which concatenates the two dataframes along either the rows or the columns.
+#
+# Notice that adding lagged samples will introduce null values for all the initial
+# samples, as there is no previous information for them.
+
+data_lagged = data_prep.skb.apply_func(get_lagged_features).drop("cnt")
+X = data_prep.skb.concat([data_lagged], axis=1).drop("cnt")
+y = data_var["cnt"].skb.mark_as_y()
+X
+
+
+# %%
+# We can use the datetime encoder defined above to observe the impact of the new
+# features on the predictions.
+
+
+vectorized = X.skb.apply(TableVectorizer(datetime=datetime_encoder))
+predictions_lagged_features = vectorized.skb.apply(default_ridge_pipeline, y=y)
+search_lagged_features = predictions_lagged_features.skb.get_grid_search(
+    fitted=True, cv=ts
+)
+search_lagged_features.detailed_results_
+
+# %%
+search_lagged_features.plot_results(min_score=0.0)
+
+# %%
+# We can see that the lagged features improved the prediction performance by a
+# large margin. Periodic features are still bringing some benefit.
+
+
+# %%
+# Plotting the prediction
+# .......................
+#
+# To have a better idea of the quality of the predictions, we can plot them directly.
+# We can use |train_test_split| to separate the data into a train split and
+# a test split.
+# In this case, we will use all the data for the year 2011 to predict year 2012.
+# The |train_test_split| function can take a custom `splitter` function that returns
+# the X and y splits based on some specific requirements; here, we specify a
+# cutoff date and use that to prepare the train an test splits.
+# For each pipeline, we run a grid search and select only the best pipeline according
+# to its test score.
+# Then, we train each pipeline on the training split and predict on the test split.
+
+
+def split_function(X, y, cutoff=None):
+    mask = X["date"] < cutoff
+    X_train, X_test = X.filter(mask), X.filter(~mask)
+    y_train, y_test = y.filter(mask), y.filter(~mask)
+    return X_train, X_test, y_train, y_test
+
+
+# %%
+split_base = predictions_base.skb.train_test_split(
```

Comment:
Could you create a function or a for loop to accumulate results in a dict? This would be easier to read